### PR TITLE
Supportmvn repository

### LINKED ISSUE #149: parse repo.maven.apache.org
https://repo.maven.apache.org/maven2/commons-collections/commons-collections/3.2.2/

parse this maven repo.

### PR DESCRIPTION
**Pull request by ctownshend**
Fixed #149, #148, did some refactoring, added test for sortIssues, added some other unit tests

test URLS: 

- https://repo.maven.apache.org/maven2/commons-collections/commons-collections/3.2.1/
- https://mvnrepository.com/artifact/org.apache.struts/struts2-core/2.2.3
- https://www.npmjs.com/package/lodash/v/4.17.9

### CODE REVIEW

**Comment by ctownshend** [comment]
Hey @DarthHater wondering if you could cast your eye over this. If you have a moment, I did some work on the legal api too. We can discuss. Sal is keen to promote it.

File: src/utils/Constants.ts [hunk 1]
```diff
@@ -46,32 +46,74 @@ export const FORMATS = {
...
+  mavencentral: 'mavencentral',
+  mvnrepository: 'mvnrepository',
+  npmJS: 'npmjs',
+  nuget: 'nuget',
+  packagist: 'packagist',
+  pypi: 'pypi',
+  rubygems: 'rubygems',
+  nexusepo: 'nexusrepo',
+  artifactory: 'artifactory'
+};
```
Comment (DarthHater) [inline src/utils/Constants.ts:93]: These are all implemented in`RewriteInReactFunZone` now, merge the newest and update, sorry, I didn't see this PR before I went and did it (add @bhamail and I as reviewers if you can when you open the PR)

File: src/utils/Constants.ts [hunk 2]
```diff
@@ -46,32 +46,74 @@ export const FORMATS = {
...
 
 export interface RepoType {
   url: string;
   repoFormat?: string;
   repoID: string;
   titleSelector?: string;
   versionPath?: string;
-  dataSource: string;
+  dataSource: string; //TODO: remove this as unnecessary now as we no longer switch datasources on RepoType
```
Comment (DarthHater) [inline src/utils/Constants.ts:101]: Yeah we should just get rid of this, I think. I wondered if it was useful to say like a format CAN get data from IQ or OSS Index though. What do you think?

File: src/utils/PageParsing.ts [hunk 3]
```diff
@@ -24,28 +24,62 @@ import {parsePyPIURL} from './PageParsing/PyPI';
...
+      return parseMavenApache(url);
+      break;
+    }
+    default: {
+      //throw undefined error
+      throw new Error(`Unhandled Repotype and URL ${repoFormat} ${url}`);
+    }
   }
-
+  // console.trace("got here by mistake")
```
Comment (DarthHater) [inline src/utils/PageParsing.ts:82]: LOL

File: src/utils/PageParsing/MavenApache.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  generatePackageURL,
+  generatePackageURLWithNamespace,
+  generatePackageURLComplete
+} from './PurlUtils';
+
+//pkg:type/namespace/name@version?qualifiers#subpath
+//Sonatype expects: "packageUrl": "pkg:maven/org.yaml/snakeyaml@1.17?type=jar"
+const parseMavenApache = (url: string): PackageURL | undefined => {
+  // console.trace("parseMavenApache", url)
```
Comment (DarthHater) [inline src/utils/PageParsing/MavenApache.ts:29]: No need to comment this out if you have the log level at trace!

File: src/utils/Constants.ts [hunk 5]
```diff
@@ -46,32 +46,74 @@ export const FORMATS = {
...
+  debian: 'deb',
+  gem: 'gem',
+  github: 'github',
+  golang: 'golang',
+  maven: 'maven',
+  npm: 'npm',
+  nuget: 'nuget',
+  pypi: 'pypi',
+  rpm: 'rpm'
+};
```
Comment (DarthHater) [inline src/utils/Constants.ts:68]: What is this for?

File: src/utils/Constants.ts (hunk 1 above)
Comment (ctownshend) [reply src/utils/Constants.ts:93]: Got it, will make sure to add you both as reviewers in future. I will merge this and fix in my PR

File: src/utils/Constants.ts (hunk 2 above)
Comment (ctownshend) [reply src/utils/Constants.ts:101]: Hmm yes, not a bad idea. Then should be a collection of some kind?

File: src/utils/PageParsing.ts (hunk 3 above)
Comment (ctownshend) [reply src/utils/PageParsing.ts:82]: :)

File: src/utils/Constants.ts (hunk 5 above)
Comment (ctownshend) [reply src/utils/Constants.ts:68]: We need to make the FORMATS the same as the PURL_TYPES. So we should drop PURL_TYPES. I will delete it.

File: src/utils/PageParsing/MavenApache.ts (hunk 4 above)
Comment (ctownshend) [reply src/utils/PageParsing/MavenApache.ts:29]: OK, will restore them

File: src/utils/Constants.ts (hunk 2 above)
Comment (DarthHater) [reply src/utils/Constants.ts:101]: Likely, we can cross that bridge in a while though. I think it's safe to leave it until then though.

**Comment by ctownshend** [comment]
I added support for #153 and it now parses the URL for search.maven.org. We still need to add the html/dom parsing for all of the relevant repos.

File: src/utils/Constants.ts (hunk 2 above)
Comment (ctownshend) [reply src/utils/Constants.ts:79]: Yes agreed for now. We'll leave it. Just a reminder to get to it when we need to.

**Review by DarthHater — APPROVED** [review]

**Pull request merged**